warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EventDataQueryExpressionVisitor.java

```diff
@@ -347,8 +402,8 @@ private static String print(JexlASTHelper.IdentifierOpLiteral iol) {
         return iol.getIdentifier() + " " + iol.getOp() + " " + iol.getLiteral();
     }
     
-    public static Set<String> extractNormalizedAttributes(AttributeFactory attrFactory, String fieldName, String fieldValue, Key key) {
-        final Set<String> normalizedAttributes = new HashSet<>();
+    public static List<Type> extractTypes(AttributeFactory attrFactory, String fieldName, String fieldValue, Key key) {
+        final List<Type> types = new ArrayList<>();
```

Comment:
Could we end up putting duplicate types into this List unintentionally (e.g: if one type may appear in multiple TypeAttributes or if a type might appear in both an AttributeBag and a TypeAtribute - we would wind up with duplicates in this list. Perhaps instead of unique types we should consider building the list of unique normalizers?

Reply:
yep, fixed